test/lint/lint-format-strings.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2018-2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#
+
+"""
+Lint format strings: This program checks that the number of arguments passed
+to a variadic format string function matches the number of format specifiers
+in the format string.
+"""
+
+import subprocess
+import re
+
+FUNCTION_NAMES_AND_NUMBER_OF_LEADING_ARGUMENTS = [
+    'FatalError,0',
+    'fprintf,1',
+    'tfm::format,1',  # Assuming tfm::::format(std::ostream&, ...
+    'LogConnectFailure,1',
+    'LogPrint,1',
+    'LogPrintf,0',
+    'printf,0',
+    'snprintf,2',
+    'sprintf,1',
+    'strprintf,0',
+    'vfprintf,1',
+    'vprintf,1',
+    'vsnprintf,1',
+    'vsprintf,1',
+    'WalletLogPrintf,0',
+]
+
+def main():
+    doctest_args = ['python3', '-m', 'doctest', 'test/lint/run-lint-format-strings.py',]
+    try:
+        subprocess.check_output(doctest_args, stderr = subprocess.STDOUT)
+    except subprocess.CalledProcessError:
+        exit(1)
+
+    for s in FUNCTION_NAMES_AND_NUMBER_OF_LEADING_ARGUMENTS:
+
+        function_name, skip_arguments = s.split(',')
+
+        matching_files_args = ['git', 'grep', '--full-name', '-l', function_name, '--','*.c', '*.cpp', '*.h']
+        try:
+            all_matching_files = subprocess.check_output(matching_files_args).decode('utf-8').splitlines()
+        except subprocess.CalledProcessError as e:
+            if e.returncode > 1: # return code is 1 when match is empty
+                exit(1)
+            all_matching_files = []
+
+        matching_files = []
+        for matching_file in all_matching_files:
+            if not re.search('^src/(leveldb|secp256k1|minisketch|tinyformat|univalue|test/fuzz/strprintf.cpp)', matching_file):
+                matching_files.append(matching_file)
+
+        run_lint_args = ['test/lint/run-lint-format-strings.py',
+                            '--skip-arguments', skip_arguments, function_name,] + matching_files
+        try:
+            subprocess.check_output(run_lint_args, stderr = subprocess.STDOUT)
+        except subprocess.CalledProcessError:
+            exit(1)
```

Comment:
nit: the convention is to use `sys.exit` instead of the built-in, which is only meant for interactive use.

Reply:
Thank you for reviewing !
I've updated the PR according to your suggestions